docs/docs/tutorials/jwt_authentication.md

```diff
@@ -0,0 +1,515 @@
+into the routes of that namespace.
+
+The usage of `bearerAuthentication` middleware, which is provided by `dart_frog_auth` is quite simple.
+We simply need to inform the a function to the `bearerAuthentication` attribute, this function receives
+the current `RequestContext` and the token that was passed in the request.
+
+If the token is valid and belongs to a user, the function must return that user. Otherwise,
+it should return null.
+
+This middleware will automatically return `unauthorized` response to incoming requests when
+no valid tokens are provided, so if we go ahead and try the following command in our terminal:
+
+```bash
+# Note the additional `-v`, so we can see the status code in the output
+curl -d "Buy bread" \
+     -v \
+     http://localhost:8080/tasks
+```
+
+We should see the following line in the output
+
+```bash
+< HTTP/1.1 401 Unauthorized
+```
+
+But if we inform a valid and correct authentication token:
+
+```bash
+curl -d "Buy bread" \
+     -v \
+     -H "Authorization: Bearer john" \
+     http://localhost:8080/tasks
+```
+
+The correct response of that route should be output:
+
+```bash
+{"recorded_task":"Buy bread"}
+```
+
+Additionally, the `bearerAuthentication` middleware will set the returned user in the request context,
+so any route handlers affected by it, will have access to the user that is currently authenticated.
+
+With that information, we can change our tasks routes to have a more interesting response:
+
+```dart
+import 'dart:io';
+
+import 'package:authenticated_app/user.dart';
+import 'package:dart_frog/dart_frog.dart';
+
+Future<Response> onRequest(RequestContext context) {
+  return switch (context.request.method) {
+    HttpMethod.post => _onPost(context),
+    _ => Future.value(Response(statusCode: HttpStatus.methodNotAllowed)),
+  };
+}
+
+Future<Response> _onPost(RequestContext context) async {
+  final task = await context.request.body();
+  final user = context.read<User>();
+
+  return Response.json(
+    body: {
+      'recorded_task': task,
+      'user_id': user.id,
+    },
+  );
+}
+```
+
+## Protecting the authentication token
+
+This is a good point to review what we have done so far:
+
+- We have created a sign in route, where credentials can be posted, and an authentication token
+  is returned if valid.
+- We have routes that can only be accessed if an authentication token is sent in the request.
+
+But like we noticed in the steps before, our authentication token is quite unsafe, it is nothing
+more than the user's username, meaning that if an ill intentioned person guesses another user's
+username, which is not something hard to do, they could pass as that user and maybe steal
+information or other bad actions.
+
+To avoid that, we have to make our tokens in a way where they cannot be faked, guessed or tampered.
+There are many ways of doing that, in this tutorial, we will use JWT, a widely used standard in the
```

Comment:
Suggestion to avoid repeating "way" again (used in the precious sentence) and "that":

```suggestion
There are many methods of doing so, in this tutorial, we will use JWT, a widely used standard in the
```